docs-client/src/containers/App/index.tsx

```diff
@@ -351,6 +392,31 @@ class App extends React.PureComponent<Props, State> {
       mobileDrawerOpen: !this.state.mobileDrawerOpen,
     });
   };
+
+  private handleCollapse(itemName: string) {
```

Comment:
`handleCollapse = (itemName) => {}` so it's bound to this

Reply:
I see `navigateTo(to: string)` does not use a fat arrow while some other methods use fat arrows. Would you mind educating me when a fat arrow should be used? I thought it's OK not to use a fat arrow for prototype methods.

Reply:
Looking again, I think `navigateTo` should use an arrow too and then we don't need an arrow in the above places where we reference it, oops. 

Unfortunately, prototype methods when used as callbacks are not bound to `this` and `this` can't be used in them, while arrow functions are. I guess we may want to follow a practice that all methods on a component class should always be arrow functions to avoid worrying about it.

Reply:
I'm curious why callback methods are not bound to `this`. Can't just the party that invokes the callback do `component.callbackMethod();` then `this` could be `component`?

Reply:
If `component` was passed around, then it works fine. But most JS APIs pass function references around, so for example you end up doing `onClick={component.callbackMethod}` (passing reference to the function, not component). As soon as you do this, for a normal prototype method it becomes unbound. One of the most frustrating aspects of Javascript (and Typescript doesn't fix it to preserve compatibility)

Reply:
I see. Thanks for explanation, @anuraaga!

Reply:
Happened to find this detailed guide too

https://github.com/Microsoft/TypeScript/wiki/'this'-in-TypeScript

Reply:
Thanks for sharing, @anuraaga. Very helpful.